tests/validator_churn.go

```diff
@@ -0,0 +1,158 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	"github.com/ava-labs/subnet-evm/core/types"
+	subnetEvmUtils "github.com/ava-labs/subnet-evm/tests/utils"
+	teleportermessenger "github.com/ava-labs/teleporter/abi-bindings/go/Teleporter/TeleporterMessenger"
+	"github.com/ava-labs/teleporter/tests/network"
+	"github.com/ava-labs/teleporter/tests/utils"
+	localUtils "github.com/ava-labs/teleporter/tests/utils/local-network-utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	. "github.com/onsi/gomega"
+)
+
+// Disallow this test from being run on anything but a local network, since it manipulates the validator set
+func ValidatorChurnGinkgo() {
+	network := &network.LocalNetwork{}
+	var (
+		teleporterMessageID *big.Int
+	)
+
+	subnets := network.GetSubnetsInfo()
+	subnetAInfo := subnets[0]
+	subnetBInfo := subnets[1]
+	teleporterContractAddress := network.GetTeleporterContractAddress()
+	fundedAddress, fundedKey := network.GetFundedAccountInfo()
+
+	subnetATeleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(
+		teleporterContractAddress, subnetAInfo.ChainRPCClient,
+	)
+	Expect(err).Should(BeNil())
+	subnetBTeleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(
+		teleporterContractAddress, subnetBInfo.ChainRPCClient,
+	)
+	Expect(err).Should(BeNil())
+
+	ctx := context.Background()
+
+	//
+	// Send a Teleporter message on Subnet A
+	//
+	log.Info("Sending Teleporter message on source chain", "destinationChainID", subnetBInfo.BlockchainID)
+	sendCrossChainMessageInput := teleportermessenger.TeleporterMessageInput{
+		DestinationChainID: subnetBInfo.BlockchainID,
+		DestinationAddress: fundedAddress,
+		FeeInfo: teleportermessenger.TeleporterFeeInfo{
+			ContractAddress: fundedAddress,
+			Amount:          big.NewInt(0),
+		},
+		RequiredGasLimit:        big.NewInt(1),
+		AllowedRelayerAddresses: []common.Address{},
+		Message:                 []byte{1, 2, 3, 4},
+	}
+
+	var receipt *types.Receipt
+	receipt, teleporterMessageID = utils.SendCrossChainMessageAndWaitForAcceptance(
+		ctx,
+		subnetAInfo,
+		subnetBInfo,
+		sendCrossChainMessageInput,
+		fundedAddress,
+		fundedKey,
+		subnetATeleporterMessenger,
+	)
+
+	sendEvent, err := utils.GetEventFromLogs(receipt.Logs, subnetATeleporterMessenger.ParseSendCrossChainMessage)
+	Expect(err).Should(BeNil())
+	sentTeleporterMessage := sendEvent.Message
+
+	// Construct the signed warp message
+	signedWarpMessageBytes := localUtils.ConstructSignedWarpMessageBytes(ctx, receipt, subnetAInfo, subnetBInfo)
+
+	//
+	// Modify the validator set on Subnet A
+	//
+
+	// Add new nodes to the validator set
+	log.Info("Adding nodes to the validator set")
+	var nodesToAdd []string
+	for i := 15; i <= 19; i++ {
```

Comment:
why do we start from 15? I believe we just have two subnets, and 5 nodes each. Can we get the total number of nodes for both subnets and start from there insted of hardcoding the value?